public/app/plugins/datasource/graphite/configuration/GraphiteDetails.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { PureComponent } from 'react';
+import { DataSourceSettings, SelectableValue } from '@grafana/data';
+import { Button, FormLabel, Select } from '@grafana/ui';
+import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
+import { GraphiteOptions, GraphiteType } from '../types';
+
+const graphiteVersions = [
+  { label: '0.9.x', value: '0.9' },
+  { label: '1.0.x', value: '1.0' },
+  { label: '1.1.x', value: '1.1' },
+];
+
+const graphiteTypes = Object.keys(GraphiteType).map((key: string) => ({
+  label: key,
+  value: (GraphiteType as any)[key],
+}));
+
+interface Props {
+  value: DataSourceSettings<GraphiteOptions>;
+  onChange: (value: DataSourceSettings<GraphiteOptions>) => void;
+}
+
+interface State {
+  showMetricTankHelp: boolean;
+  graphiteVersion: string;
+}
+
+export class GraphiteDetails extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      showMetricTankHelp: false,
+      graphiteVersion: '',
+    };
+  }
+
+  componentDidMount() {
+    const { value } = this.props;
+    if (!value.id) {
+      this.setState({ graphiteVersion: '' });
+    }
+
+    getDatasourceSrv()
+      .loadDatasource(value.name)
+      .then((dataSource: any) => {
+        return dataSource.getVersion();
+      })
+      .then((version: any) => {
+        if (!version) {
+          return;
+        }
+        this.setState({ graphiteVersion: version });
```

Comment:
This may happen when the component is already unmounted. @shavonn recently solved it with `CancelablePromise`, but it shows a need for us to come up with Redux recommendation for handling async in components. WDYT @peterholmberg about moving this logic to Redux? This would also create a reference implementation for other datasource configs migrations.  

cc @hugohaggmark